src/plans/details/tabs/Mappings/components/PlanNetworkMapEdit/components/PlanNetworkMapFieldsTable.tsx

```diff
@@ -0,0 +1,158 @@
+          oVirtNicProfiles,
+          usedSourceNetworks,
+          values,
+          vms,
+        }),
+    },
+  });
+
+  // validate initial values on first load
+  useEffect(() => {
+    setTimeout(async () => {
+      await trigger();
+    }, 0);
+  }, [trigger]);
+
+  return (
+    <FieldBuilderTable
+      headers={[
+        {
+          label: networkMapFieldLabels[NetworkMapFieldId.SourceNetwork],
+          width: 45,
+        },
+        {
+          label: networkMapFieldLabels[NetworkMapFieldId.TargetNetwork],
+          width: 45,
+        },
+      ]}
+      fieldRows={networkMappingFields.map((field, index) => ({
+        ...field,
+        inputs: [
+          <SourceNetworkField
+            fieldId={getNetworkMapFieldId(NetworkMapFieldId.SourceNetwork, index)}
+            key={getNetworkMapFieldId(NetworkMapFieldId.SourceNetwork, index)}
+            usedSourceNetworks={usedSourceNetworks}
+            otherSourceNetworks={otherSourceNetworks}
+          />,
+          <TargetNetworkField
+            fieldId={getNetworkMapFieldId(NetworkMapFieldId.TargetNetwork, index)}
+            key={getNetworkMapFieldId(NetworkMapFieldId.TargetNetwork, index)}
+            targetNetworks={targetNetworks}
+          />,
+        ],
+      }))}
+      addButton={{
+        isDisabled:
+          [...usedSourceNetworks, ...otherSourceNetworks].length === networkMappingFields.length ||
+          isLoading ||
+          Boolean(loadError),
+        label: t('Add mapping'),
+        onClick: async () => {
+          const missingNetwork = usedSourceNetworks.find(
+            (sourceNetwork) =>
+              !networkMappingFields.some(
+                (netMapping) => netMapping.sourceNetwork.id === sourceNetwork.id,
+              ),
+          );
+
+          append({
+            [NetworkMapFieldId.SourceNetwork]:
+              missingNetwork ?? defaultNetMapping[NetworkMapFieldId.SourceNetwork],
+            [NetworkMapFieldId.TargetNetwork]: hasPodNetworkMappings(networkMappings)
+              ? ignoreNetMapping[NetworkMapFieldId.TargetNetwork]
+              : defaultNetMapping[NetworkMapFieldId.TargetNetwork],
+          });
+
+          await trigger();
+        },
+      }}
+      removeButton={{
+        isDisabled: (index) => {
+          if (networkMappingFields.length <= 1) {
+            return true;
+          }
+          return usedSourceNetworks.some(
+            (network) => network.id === networkMappingFields[index].sourceNetwork.id,
+          );
+        },
+        onClick: (index) => {
+          if (
+            networkMappingFields.length > 1 &&
+            !usedSourceNetworks.some(
+              (network) => network.id === networkMappingFields[index].sourceNetwork.id,
+            )
+          ) {
+            remove(index);
+          }
+        },
+        tooltip: (index) => {
+          if (networkMappingFields.length <= 1) {
+            return t('At least one network mapping must be provided.');
```

Comment:
Why do you enforce at least one mapping? Can't we have a plan without any mapping (if no network or storage exists)?

Reply:
you mean we should allow an empty mapping?